tests/phpunit/bootstrap.php

```diff
@@ -29,9 +25,14 @@
 	$_test_root = '/tmp/wordpress-tests-lib';
 }
 
-// Force plugin to be active.
-$GLOBALS['wp_tests_options'] = array(
-	'active_plugins' => array( basename( TESTS_PLUGIN_DIR ) . '/plugin.php' ),
+require_once $_test_root . '/includes/functions.php';
+
+tests_add_filter(
+	'plugins_loaded',
+	static function (): void {
+		require_once TESTS_PLUGIN_DIR . '/plugin.php';
+	},
+	1
 );
```

Comment:
This is not great, as it would be loading the plugin too late.

What problem exactly does it solve? If it's needed for a specific setup, we should make it conditional only if the more appropriate approach (directly filtering `active_plugins` option) does not work. That way, if this ever leads to a problem, at least it wouldn't affect the overall tests infrastructure for everyone.

Reply:
IIRC I changed it mainly because this is the common way to set up plugin tests. But reverting that change actually seems to address the remaining two failures, so that's nice :)

Reply:
> because this is the common way to set up plugin tests

Curious where you get it from - I've never seen it :)

Reply:
Any project that uses `wp scaffold` or a derivation of that setup, see https://github.com/wp-cli/scaffold-command/blob/036c5152f85861bbd46ec63b36cde9074d786117/templates/plugin-bootstrap.mustache#L28-L35

Reply:
Hmm, FWIW I think this should be fixed there as it leads to problems - evidenced by this PR.

I don't see any reason not to filter the plugin option, this is something the WordPress test suite specifically allows for this kind of purpose with the `wp_tests_options` global.

Reply:
I just looked at the code, but it's using `muplugins_loaded` - so this is different from what you had changed it to here.

Using `muplugins_loaded` is okay, because that still loads the plugin early enough - even earlier than would be normal. But that's better than too late because everything that a plugin _could_ do would still work.

I still think filtering `active_plugins` is the even better solution, but since it's using `muplugins_loaded`, that's fine. I was thrown off only because here the code was using `plugins_loaded`.

Reply:
If it needs to be as early as possible, how about a new dedicated drop-in plugin file in WordPress core itself?

(Sorry if I've mentioned this idea elsewhere. Having deja vu!)